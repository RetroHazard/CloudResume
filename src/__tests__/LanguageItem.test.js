import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import LanguageItem from '../components/language_item';

// Mock the DataLoader component
jest.mock('../utils/dataLoader', () => {
    return ({ file, children }) => {
        const mockData = {
            language_skills: [
                { flag: 'emojione:flag-for-united-states', language: 'English', level: 'Native' },
                { flag: 'twemoji:flag-for-france', language: 'French', level: 'Intermediate' },
                { flag: 'emojione:flag-for-spain', language: 'Spanish', level: 'Beginner' },
            ],
        };
        return children(mockData);
    };
});

test('LanguageItem renders correctly and matches snapshot', () => {
    const { asFragment } = render(<LanguageItem />);
    expect(asFragment()).toMatchSnapshot();
});
