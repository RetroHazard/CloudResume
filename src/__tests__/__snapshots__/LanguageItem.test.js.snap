// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageItem renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex gap-3 flex-wrap"
  >
    <div
      class="skill-block"
    >
      <iconify-icon
        height="1.25em"
        icon="emojione:flag-for-united-states"
        inline="true"
        width="1.25em"
      />
      English - Native
    </div>
    <div
      class="skill-block"
    >
      <iconify-icon
        height="1.25em"
        icon="twemoji:flag-for-france"
        inline="true"
        width="1.25em"
      />
      French - Intermediate
    </div>
    <div
      class="skill-block"
    >
      <iconify-icon
        height="1.25em"
        icon="emojione:flag-for-spain"
        inline="true"
        width="1.25em"
      />
      Spanish - Beginner
    </div>
  </div>
</DocumentFragment>
`;
